Plugins/NativeCamera/Source/Android/NativeCameraImpl.cpp

```diff
@@ -51,35 +70,92 @@ namespace Babylon::Plugins
         return oesTexture;
     }
 
-    std::string Camera::Impl::GetCameraId(bool frontCamera)
+    Camera::Impl::CameraConfiguration Camera::Impl::GetCameraId(uint32_t maxWidth, uint32_t maxHeight, bool frontCamera)
     {
+        // Get the list of available cameras
         API24::ACameraIdList *cameraIds = nullptr;
         GET_CAMERA_FUNCTION(ACameraManager_getCameraIdList)(m_cameraManager, &cameraIds);
 
-        std::string cameraId{};
+        const char* bestCameraId{nullptr};
+        uint32_t bestPixelCount{0};
+        uint32_t bestDimDiff{0};
+        uint32_t bestWidth{0};
+        uint32_t bestHeight{0};
+        int32_t bestSensorOrientation{0};
+        bool foundExactMatch{false};
 
+        // Iterate over all of the cameras and find the one that has the best stream configuration
         for (int i = 0; i < cameraIds->numCameras; ++i)
         {
-            const char *id = cameraIds->cameraIds[i];
+            const char* id = cameraIds->cameraIds[i];
```

Comment:
```suggestion
            const char* id{cameraIds->cameraIds[i]};
```

Reply:
I have been following [this guideline](https://stackoverflow.com/questions/9976927/when-to-use-the-brace-enclosed-initializer#:~:text=If%20the%20values%20you%20are%20initializing%20with%20are,complex%20number%29%2C%20use%20curly%20braces%20initialization%20if%20available.) recently for brace initialization. I don't think we should just blindly use brace initialization.

Reply:
😫

Reply:
Since I was only changing the whitespace on this line anyway I'll leave it as the `=` notation